### DISCUSSION TITLE
Where clause doesn't work with column masking (with Ranger)

### DISCUSSION THREAD

**Discussion by vijay-jangir** [trinodb/trino#22157] [Q&A]
We're using ranger with Trino, 
WHen we implement masking policy on a column, we're not abel to put filter on the same column.

Let's say i have column masking on phone_number, i am not able to filter records of only 1 phone number.

**Comment by sajjoseph** [comment]
Please provide more details including sample query and results (including exceptions), versions of software (Trino, ranger ...) etc.

**Reply by vijay-jangir** [reply]
we're using trino 437
ranger 2.4
Sample query. 

Let's say phone_number is masked in employee table.
and there's another table mobile_bills, which has phone_number, amount

I want to do a query
select eid, phone_number, amount from employee t1 inner join mobile_bills t2 on t1.phone_bumber = t2.phone_number;

or
select eid where phone_number = "123-123-123";

both the queries doesn't work, as the filter or join are applied after masking is applied. 
@sajjoseph
